chainer/links/connection/lstm.py

```diff
@@ -7,7 +7,75 @@
 from chainer import variable
 
 
-class LSTM(link.Chain):
+class LSTMBase(link.Chain):
+
+    def __init__(self, in_size, out_size,
+                 lateral_init=None, upward_init=None,
+                 bias_init=0, forget_bias_init=0):
+        super(LSTMBase, self).__init__(
+            upward=linear.Linear(in_size, 4 * out_size, initialW=0),
+            lateral=linear.Linear(out_size, 4 * out_size,
+                                  initialW=0, nobias=True),
+        )
+        self.state_size = out_size
+
+        for i in six.moves.range(0, 4 * out_size, out_size):
+            initializers.init_weight(
+                self.lateral.W.data[i:i + out_size, :], lateral_init)
+            initializers.init_weight(
+                self.upward.W.data[i:i + out_size, :], upward_init)
+
+        a, i, f, o = lstm._extract_gates(
+            self.upward.b.data.reshape(1, 4 * out_size, 1))
+        initializers.init_weight(a, bias_init)
+        initializers.init_weight(i, bias_init)
+        initializers.init_weight(f, forget_bias_init)
+        initializers.init_weight(o, bias_init)
+
+
+class StatelessLSTM(LSTMBase):
+
+    """Stateless LSTM layer.
+
+    This is a fully-connected LSTM layer as a chain. Unlike the
+    :func:`~chainer.functions.lstm` function, this chain holds upward and
+    lateral connections as child links. This link doesn't keep cell and
+    hidden states.
+
+    Args:
+        in_size (int): Dimensionality of input vectors.
+        out_size (int): Dimensionality of output vectors.
+
+    Attributes:
+        upward (chainer.links.Linear): Linear layer of upward connections.
+        lateral (chainer.links.Linear): Linear layer of lateral connections.
+
+    """
+
+    def __call__(self, c, h, x):
+        """Updates the internal state and returns the LSTM outputs.
```

Comment:
This summary might be confusing a bit, since there is no internal state in StatelessLSTM.